R/GeospatialFunctions.R

```diff
@@ -0,0 +1,571 @@
+                               popup = paste0("NHDPlus HR Catchment ID: ", nearby_catchments$ATTAINS.nhdplusid)),
+        silent = TRUE)
+    
+    # Add ATTAINS polygon features (if they exist):
+    try(map <- map %>%
+          leaflet::addPolygons(data = polygons_mapper,
+                               color = ~polygons_mapper$col,
+                               fill = ~polygons_mapper$col,
+                               weight = 3, fillOpacity = 1,
+                               popup = paste0("Assessment Unit Name: ", polygons_mapper$assessmentunitname, 
+                                              "<br> Assessment Unit ID: ", polygons_mapper$assessmentunitidentifier,
+                                              "<br> Status: ", polygons_mapper$overallstatus,
+                                              "<br> Assessment Unit Type: ", polygons_mapper$type,
+                                              "<br> <a href=", polygons_mapper$waterbodyreportlink, " target='_blank'>ATTAINS Link</a>")),
+        silent = TRUE)
+    
+    # Add ATTAINS lines features (if they exist):
+    try(map <- map %>%
+          leaflet::addPolylines(data = lines_mapper,
+                                color = ~lines_mapper$col,
+                                weight = 4, fillOpacity = 1,
+                                popup = paste0("Assessment Unit Name: ", lines_mapper$assessmentunitname, 
+                                               "<br> Assessment Unit ID: ", lines_mapper$assessmentunitidentifier,
+                                               "<br> Status: ", lines_mapper$overallstatus,
+                                               "<br> Assessment Unit Type: ", lines_mapper$type,
+                                               "<br> <a href=", lines_mapper$waterbodyreportlink, " target='_blank'>ATTAINS Link</a>")),
+        silent = TRUE)
+    
+    # Add ATTAINS point features (if they exist):
+    try(map <- map %>%
+          leaflet::addCircleMarkers(data = points_mapper, 
+                                    lng = ~LongitudeMeasure, lat = ~LatitudeMeasure, 
+                                    color = ~points_mapper$col, fillColor = ~points_mapper$col, 
+                                    fillOpacity = 1, stroke = TRUE, weight = 1.5, radius = 5, 
+                                    popup = paste0("Assessment Unit Name: ", points_mapper$assessmentunitname, 
+                                                   "<br> Assessment Unit ID: ", points_mapper$assessmentunitidentifier,
+                                                   "<br> Status: ", points_mapper$overallstatus,
+                                                   "<br> Assessment Unit Type: ", points_mapper$type,
+                                                   "<br> <a href=", points_mapper$waterbodyreportlink, " target='_blank'>ATTAINS Link</a>")),
+        silent = TRUE)
+    
+    # Add WQP observation features (should always exist):
+    try(map <- map %>%
```

Comment:
Since there should always be WQP data would it make sense to remove `try()`? You could have a check that creates an error if there's no WQP data if that would indicate a problem with the dataset.